### DISCUSSION TITLE
blog/2023-02-06-boost-your-power-with-baseline-covariates/

### DISCUSSION THREAD

**Discussion by giscus** [ASKurz/blogdown5#5] [Comments]
# blog/2023-02-06-boost-your-power-with-baseline-covariates/

Part 1 of the GLM and causal inference series.

https://timely-flan-2986f4.netlify.app/blog/2023-02-06-boost-your-power-with-baseline-covariates/

**Comment by ASKurz** [comment]
Y'all can add your comments in this section. You'll note these support Markdown goodies, like *italic* and **boldface** text. They also support $\LaTeX$ equations, like $y = mx + b$.

**Comment by isabellaghement** [comment]
Nice post, Solomon! Here are a few minor comments for your consideration. 

1.  The mention of “true experiments” at the beginning of the post made me wonder if there is such a thing of “false experiments”. (: 

Maybe a clarification is needed here - are these experiments which rely on randomization to assign subjects to treatments? 

2. The research question currently reads: “Which group is better for weight-loss?”, which doesn’t seem ideal to me. 

At a minimum, perhaps this could be revised: “Which treatment is better for weight loss?” (Would anyone expect placebo or control to be better than all others?) 

Even better, you could (re)state the research question after restricting yourself to just 2 groups: “Is the experimental treatment better for weight loss than the wait-list control treatment?”

3. An upfront mention of the fact that you will use “causal language” to present your findings would be helpful: “the experimental treatment results in greater weight loss on average than the control” (as opposed to “is associated with”). 

4. Can you briefly explain why you started with a model which omits pre as a covariate? Is it because you’d expect randomization to help you make a fair comparison between treatment groups even if you don’t adjust for pre?  (A brief mention as to why analyzing post - pre is not such a good idea may be helpful.) 

5. You have to excuse my causal inference naivety, but isn’t true that first and foremost what helps us make the comparison of interest between the experimental and control treatments is randomizing the subjects to the two groups? (We hope that randomization would make the two groups comparable with respect to all other relevant characteristics which could affect their response to treatment.)  Then, adjusting for covariates like pre in our modelling will help us get a more precise estimate of the treatment effect? (Though see my next comment - isn’t it a more precise estimate of a treatment effect that has a different meaning?) 

5. I am confused about what our target effect for inference is from a causal perspective (which may be a side effect to my naivety).  My non-causal knowledge tells me that the model *without* pre as a covariate helps us estimate the treatment effect among women represented by the one in the experiment, regardless of their pre value. By contrast, the model *with* pre as a covariate helps us estimate the treatment effect among women represented by the one on the experiment who have the same value of pre.  Not sure whether in causal inference these two effects are still assumed to be different things? If they are, maybe a clarification is warranted that the ANOVA and ANCOVA models are estimating the treatment effect but in different populations of women?  

The ANCOVA model is “zooming in” on just women with the same value for pre who either received the experimental treatment or the control treatment and contrasting them in terms of mean weight.  So it targets a “special” version of the treatment effect. 

6. Maybe a mention that we are placing a lot of hope on randomization to act as the “great equalizer” might be useful? When so many other things can affect the response to treatment (of which pre is just one!), maybe it’s understandable that ran

**Reply by isabellaghement** [reply]
The last (incomplete) item should be labelled as 7 and should conclude like this:

… maybe it’s understandable that randomization may work well in balancing the distribution of some but not all of these other things.  

Which is where adjusting for baseline covariates comes in handy - it can mop up the “unbalanced” bits left over by randomization.

**Reply by ASKurz** [reply]
WRT (1), are y'all statisticians not used to using the language of “true experiments?” I was under the impression this was part of the general nomenclature. 

WRT (4), the tension between models without and with covariates is central to the methodological issues to come in future posts. But perhaps stepping back a little, the model without the covariates is the simplest model a researcher can use to compare the two groups. It's the analogue to the $t$-test and it's what I believe many of my applied colleagues would consider the default. From my applied perspective, I can't imagine why one wouldn't start with the model without covariates.

WRT (5), yes that's my general understanding and it's my presumption my audience would come with that as background knowledge, even if they wouldn't be able to express it with rigorous methodological language.

**Comment by Dpananos** [comment]
A few comments:

* Visualizations -- aside from the pairs of histograms -- are notably absent from this post.  Its a real shame because your claim that "pre-treatment measurements tend to have strong correlations with post-treatment measurements" could be very easily demonstrated.

* Is it worth discussing why the correlation between pre and post shrinks the standard error?  IMO because you have so few variables in this regression, you could very easily: plot pre against post, compute residuals, and discuss how the error term shrinks due to the correlation between pre and post.  Since the error term appears in the standard error, this has the effect of increasing precision.

* I'm not saying this is needed, the post as it stands is very good.  But I'm wondering if it is worth it to go the extra mile and answer the "why" behind the shrinking standard error.  IMO it is worth it, because while we may not have pre-measurements in all experiments, we may be able to record other variables which ARE highly correlated with the outcome and have the same effect.  Pre-measurements are perhaps the most notable kinds of covariates to do this with.

**Reply by Dpananos** [reply]
Case in point, plot

```
horan1971 %>% 
  filter(treatment %in% c("delayed", "experimental")) %>% 
  ggplot(aes(pre, post)) + 
  geom_point() +
  theme(aspect.ratio = 1)
```

and you can just *see* that the residual variation is smaller than the marginal variation, and it looks like experimental units have lower post weight on average.  That's the benefit of reducing variation through other correlated variables.

**Reply by Dpananos** [reply]
and here is how we might plot the residual variance vs the marginal variance of post

```

horan1971 %>% 
  mutate(postc = post - mean(post)) %>% 
  select(treatment, resid, postc) %>% 
  pivot_longer(cols = c('resid', 'postc'), names_to = 'which', values_to = 'y') %>% 
  ggplot(aes(y)) + 
  geom_histogram() +
  facet_wrap(~which, nrow=2)
```

Thinking of the marginal mean as a prediction allows us to think of centered post weight `postc` as a kind of residual.  Note that the two residuals here have much smaller variance. THATS IMPROVED PRECISION!  That's the "why"

**Reply by ASKurz** [reply]
Where is your `resid` column coming from?

**Reply by Dpananos** [reply]
Ah shit, sorry.  Here is the full code

```
library(tidyverse)

horan1971 <- tibble(
  sl = c(letters[1:22], letters[1:20], letters[1:19], letters[1:19]),
  sn = 1:80,
  treatment = factor(rep(1:4, times = c(22, 20, 19, 19))),
  pre = c(149.5, 131.25, 146.5, 133.25, 131, 141, 145.75, 146.75, 172.5, 156.5, 153, 136.25, 148.25, 152.25, 167.5, 169.5, 151.5, 165, 144.25, 167, 195, 179.5,
          127, 134, 163.5, 155, 157.25, 121, 161.25, 147.25, 134.5, 121, 133.5, 128.5, 151, 141.25, 164.25, 138.25, 176, 178, 183, 164,
          149, 134.25, 168, 116.25, 122.75, 122.5, 130, 139, 121.75, 126, 159, 134.75, 140.5, 174.25, 140.25, 133, 171.25, 198.25, 141.25,
          137, 157, 142.25, 123, 163.75, 168.25, 146.25, 174.75, 174.5, 179.75, 162.5, 145, 127, 146.75, 137.5, 179.75, 168.25, 187.5, 144.5),
  post = c(149, 130, 147.75, 139, 134, 145.25, 142.25, 147, 158.25, 155.25, 151.5, 134.5, 145.75, 153.5, 163.75, 170, 153, 178, 144.75, 164.25, 194, 183.25,
           121.75, 132.25, 166, 146.5, 154.5, 114, 148.25, 148.25, 133.5, 126.5, 137, 126.5, 148.5, 145.5, 151.5, 128.5, 176.5, 170.5, 181.5, 160.5,
           145.5, 122.75, 164, 118.5, 122, 125.5, 129.5, 137, 119.5, 123.5, 150.5, 125.75, 135, 164.25, 144.5, 135.5, 169.5, 194.5, 142.5,
           129, 146.5, 142.25, 114.5, 148.25, 161.25, 142.5, 174.5, 163, 160.5, 151.25, 144, 135.5, 136.5, 145.5, 185, 174.75, 179, 141.5)) %>% 
  mutate(treatment = factor(treatment, labels = c("delayed", "placebo", "scheduled", "experimental")))



mod <- lm(post ~ pre + treatment, data=horan1971)


horan1971$resid <- resid(mod)


horan1971 %>% 
  mutate(postc = post - mean(post)) %>% 
  select(treatment, resid, postc) %>% 
  pivot_longer(cols = c('resid', 'postc'), names_to = 'which', values_to = 'y') %>% 
  ggplot(aes(y)) + 
  geom_histogram() +
  facet_wrap(~which, nrow=2)
```

@ASKurz what do you think?

**Reply by ASKurz** [reply]
Ah, I believe I follow. Why would one need to retain `treatment` within the `select()` part of the code, though?

**Reply by Dpananos** [reply]
You don't, I just did it because I was going to look at treatment/control but realized I already adjusted for it in the model. You can leave it out safely

**Reply by isabellaghement** [reply]
For plotting histograms in this blog post, can I just suggest that we can greatly improve the overall appearance of the histograms by doing these things: 

1. Use the option **color = white** inside geom_histogram() (but outside its aes()); 


2. Use a binwidth different than the default; see the very bottom of the help page for geom_histogram for an example of how to do this; 

3. Add colour by using the option **fill = “dodgerblue”** inside geom_histogram() (but outside its aes());

4. Ditch that horrendous grey background theme in favour of a modern theme with a white background by adding theme_minimal() at the end of the code producing the histogram.

**Reply by ASKurz** [reply]
I actually want to keep the visuals streamlined in this series. To my mind, there's enough heavy lifting going on with the theoretical and computational side. Plus, I'm partial to many of the **ggplot2** defaults, like the gray backgrounds.

**Reply by Dpananos** [reply]
Obviously your call, just wanted to highlight the opportunity to build intuition and answer a "why"

**Reply by ASKurz** [reply]
@Dpananos, your response makes me think I should clarify that my comment about keeping the visuals streamlined was with reference to @isabellaghement's suggestions about aesthetics. I want to keep the overall aesthetics pretty close to how they're currently presented. But WRT your broader point about the histogram showing the residuals versus the observed data, I'm still chewing on that one. My main concern is it's not obvious to me that that visual will clarify why the covariate will reduce the standard error. I do take your overall point that the post might do a better job explaining the *Why* part, I'm just not sure that's the best way to do it.

**Reply by isabellaghement** [reply]
Well, at least I can show you what I mean, Solomon, when I say that the histograms could be improved.  They do look a tad boring as it stands, but that may be a matter of personal taste.  Since many of us are visual learners, it doesn't hurt to make the visuals "pop" a bit more.  However, I totally understand that that is not where you want to invest your efforts; in my mind, there is a difference between "streamlined" and "exciting".  (: 

![image](https://user-images.githubusercontent.com/57458528/219812668-65ace9fb-a7d9-4afc-bcb1-62364c1e8853.png)

Here is the R code I used - I think a binwidth of 5 (currently used) is too small and using a binwidth of 10 or even 15 would be more appropriate for these data.   I picked the colours from one of the discrete colour palettes mentioned on this website (i.e., dichromat::BluetoOrange_8): 

[https://r-charts.com/color-palettes/](https://r-charts.com/color-palettes/)

```
library(magrittr)
library(dplyr)
library(ggplot2)

solomon <-
horan1971 %>%  
  ggplot(aes(x = post)) +
  geom_histogram(binwidth = 5) +
  xlab("post-treatment weight (lbs)") +
  facet_wrap(~ treatment, labeller = label_both)

solomon 


bw <- function(x) 2 * IQR(x) / (length(x)^(1/3))


isabella <- 
  horan1971 %>%  
  ggplot(aes(x = post, fill = treatment)) +
  geom_histogram(binwidth = 10, 
                 colour = "white", 
                 alpha = 1, 
                 show.legend = FALSE) +
  xlab("post-treatment weight (lbs)") +
  facet_wrap(~ treatment, labeller = label_both) + 
  scale_fill_manual(values = c("#4CC4FF","#FFC44C")) + 
  theme_minimal()

isabella 

library(patchwork)

solomon + ggtitle("solomon") + 
isabella + ggtitle("isabella") + 
 plot_layout(nrow = 2)


```

**Comment by sjwild** [comment]
Overall, good post. I have a few comments in addition to what the others have said:
* I agree with @Dpananos about visuals. Especially for an audience not inducted into the world of stats and jargon, visuals are how they'll make sense of everything. Because you said your audience would likely start with ANOVA, it seems prudent to include more visuals.
* Regarding why adding a covariate will shrink the standard error, plots are very helpful. A good example comes from _The Effect_, where it's used to show how regression works. See figure 4.10 [here](https://theeffectbook.net/ch-DescribingRelationships.html) for a good example of how it can help. I think a few paragraphs or a visual for the *why* is worth it for the intuition in builds.
* The use of residuals in Figure 4.10 links to something in covered often in econometrics, the Frisch-Waugh-Lovell theorem. It's not even remotely necessary for this post, and I include it because you can see the explanation for how it deals with the residuals in the same chapter of _The Effect_ if you search for "Frisch-Waugh-Lovell".
* I doubt it's something you want to get into for this post, but there is a long-running discussion about ANOVA vs. ANCOVA for this type of data in terms of causal inference. See, eg., Senn's post [here](https://errorstatistics.com/2019/08/02/s-senn-red-herrings-and-the-art-of-cause-fishing-lords-paradox-revisited-guest-post/). This links to (6) from @isabellaghement.  
* I like to footnote about change scores. Change scores suck.
* I agree with (4) from @isabellaghement. My suggestion is to include a paragraph stating that most researchers might consider starting with ANOVA, but that it is the wrong approach for reasons to come (or just for reasons).


Here is some code showing how visualizing residuals could help:
```r

# without adjusting for prec, experimental needs to explain differences in weight. We can clearly see # the relationship between post and prec, and experimental needs to try explain that relationship as well
# As a result, the math can't distinguish between the contributions of prec and experimental. 
# Results? Wide CIs
horan1971 %>%
  ggplot() +
  geom_point(mapping = aes(x = prec, 
                           y = post,
                           colour = as.factor(experimental)))

# Remove relationship between prec and post.
# Also regress experimental on prec, for FWL at the end.
horna1971$r1 <- residuals(lm(experimental ~ prec, data = horan1971)) 
horan1971$r2 <- residuals(lm(post ~ prec, data = horan1971))


# Now of effect of prec has been pulled out, so experimental doesn't need to explain it. 
# no clear pattern in residuals.
horan1971 %>%
  ggplot() +
  geom_point(mapping = aes(x = r2, 
                           y = post,
                           colour = as.factor(experimental)))


# FWL, just because
FWL <- lm(r2 ~ 0 + r1, data = horan1971) # coef should match coef for experimental in ols2 and ols3

FWL
```

**Reply by sjwild** [reply]
Smarter me should have shown it this way

![lots of variability to explain](https://raw.githubusercontent.com/sjwild/misc/main/lots_of_variability.png)
![less variability to explain](https://raw.githubusercontent.com/sjwild/misc/main/less_variability.png)


```r
mean_post <- mean(horan1971$post)
ymin <- min(horan1971$post - mean_post) - 5
ymax <- max(horan1971$post - mean_post) + 5


horan1971 %>%
  ggplot() +
  geom_point(mapping = aes(x = as.factor(experimental), 
                           y = post - mean_post,
                           colour = as.factor(experimental)),
             show.legend = FALSE) +
  labs(title = "Look how much variability experimental must explain",
       y = "Weight to explain",
       x = "Treatment") +
  ylim(ymin, ymax)

horan1971 %>%
  ggplot() +
  geom_point(mapping = aes(x = as.factor(experimental), 
                           y = r2,
                           colour = as.factor(experimental)),
             show.legend = FALSE) +
  labs(title = "Much less variability for experimental to explain",
       y = "Weight to explain",
       x = "Treatment") +
  ylim(ymin, ymax)
```

**Comment by chelseaparlett** [comment]
- Silly issue: you called participants "particulates" at one point ahahah

- You also talk about how we can call information based interventions a "placebo". I wonder if the terminology "active control" is relevant (or is this not active enough) would help contextualize this for psych people?

- You do a GREAT JOB explaining why mean centering helps interpretation in ANCOVA.
I know this isn't the point of the post, but I wondered why you started with the simple ANOVA on post test and compared it to ANCOVA. Typically people compare ANCOVA to a model predicting difference scores.

- The plot of the two effects with CI's is just gorgeous.

**Reply by chelseaparlett** [reply]
lol nevermind about the change score model, I didn't read your footnote hahahah

**Reply by ASKurz** [reply]
Thank you for the kind feedback, @cmparlettpelleriti. Dont' sleep on my footnotes, though. There's gold in them footnotes, @cmparlettpelleriti. Gold!
